drracket-test/tests/drracket/test-engine-test.rkt

```diff
@@ -431,34 +433,48 @@
                (expected got)]))]
 	 [check-failures
 	  (lambda (where signature-violations-expected check-failures-expected)
-	    (let ((text
-		   (cond
-		    ((send (send definitions-text get-tab) get-test-editor)
-		     => (lambda (test-editor)
-			  (let ((text (send test-editor get-text 0 'eof #t)))
-			    (if (string=? text "")
-				#f
-				text))))
-		    (else #f))))
-	   
-	      (cond
-	       ((and (null? signature-violations-expected)
-		     (null? check-failures-expected))
-		(when text
-		  (eprintf "FAILED: ~s ~s expected ~s to produce no check failures or signature violations:\ngot:\n~a\ninstead\n"
-			   where (language) expression text)))
-	       (text
-		(let-values (((test-count test-passed-count signature-violation-count check-failures signature-violations)
-			      (parse-test-failures text)))
-		  (when (not (equal? check-failures check-failures-expected))
-		    (eprintf "FAILED: ~s ~s expected ~s to produce check failures:\n~s\ngot:\n~s\ninstead\n"
-			     where (language) expression check-failures-expected check-failures))
-		  (when (not (equal? signature-violations signature-violations-expected))
-		    (eprintf "FAILED: ~s ~s expected ~s to produce signature violations:\n~s\ngot:\n~s\ninstead\n"
-			     where (language) expression signature-violations-expected signature-violations))))
-	       (else
-		(eprintf "expected ~a check failures and ~a signature violations but got none"
-			 (length check-failures-expected) (length signature-violations-expected))))))]
+	    (define text
+       (cond
+         [(send (send definitions-text get-tab) get-test-editor)
+          =>
+          (lambda (test-editor)
+            (let ([text (send test-editor get-text 0 'eof #t)]) (if (string=? text "") #f text)))]
+         [else #f]))
+     
+     (cond
+       [(and (null? signature-violations-expected) (null? check-failures-expected))
+        (when text
+          (eprintf
+           "FAILED: ~s ~s expected ~s to produce no check failures or signature violations:\ngot:\n~a\ninstead\n"
+           where
+           (language)
+           expression
+           text))]
+       [text
+        (let-values ([(test-count test-passed-count
```

Comment:
This formats poorly :(

I'm surprised it's not converted to `define-values` though.

Reply:
There's 20 fixes in this pull request already, so it won't add more.